src/Plugin/Field/FieldFormatter/MultiMedia.php

```diff
@@ -0,0 +1,435 @@
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+      // Declare a setting named 'text_length', with
+      // a default value of 'short'
+      'image_formatter' => 'responsive',
+      'image_formatter_option' => 'large',
+      'video_formatter' => 'oembed',
+      'video_formatter_option' => 'default',
+      'video_formatter_height' => 0,
+      'video_formatter_width' => 0,
+    ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+
+    $form['image_formatter'] = [
+      '#title' => $this->t('Image Formatter'),
+      '#type' => 'select',
+      '#options' => [
+        'responsive' => $this->t('Media Responsive Image Style '),
+        'static' => $this->t('Media Image Style'),
+        'rendered' => $this->t('Rendered entity'),
+      ],
+      '#default_value' => $this->getSetting('image_formatter'),
+    ];
+
+    $form['image_formatter_option'] = [
+      '#title' => $this->t('Image Formatter Setting'),
+      '#type' => 'textfield',
+      '#default_value' => $this->getSetting('image_formatter_option'),
+      '#description' => $this->t('The machine name key for the display mode, style, etc.'),
+    ];
+
+    $form['video_formatter'] = [
+      '#title' => $this->t('Video Formatter'),
+      '#type' => 'select',
+      '#options' => [
+        'oembed' => $this->t('oEmbed Content'),
+        'rendered' => $this->t('Rendered entity'),
+      ],
+      '#default_value' => $this->getSetting('video_formatter'),
+    ];
+
+    $form['video_formatter_option'] = [
+      '#title' => $this->t('Video Formatter Setting'),
+      '#type' => 'textfield',
+      '#default_value' => $this->getSetting('video_formatter_option'),
+      '#description' => $this->t('The machine name key for the display mode, style, etc.'),
+    ];
+
+    $form['video_formatter_height'] = [
+      '#title' => $this->t('Video Max Height (px)'),
+      '#type' => 'textfield',
+      '#default_value' => $this->getSetting('video_formatter_height'),
+      '#description' => $this->t('Integer value for max height.'),
+    ];
+
+    $form['video_formatter_width'] = [
+      '#title' => $this->t('Video Max Width (px)'),
+      '#type' => 'textfield',
+      '#default_value' => $this->getSetting('video_formatter_width'),
+      '#description' => $this->t('Integer value for max width.'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+    foreach ($items as $delta => $item) {
+    // Get the media item.
+      $media_id = $item->getValue()['target_id'];
+      $media_item = Media::load($media_id);
+      $type = $media_item->bundle();
+
+      // Depending on the type of media referenced
+      switch ($type) {
+        case 'image':
```

Comment:
The bundle name might not be like this. the bundle could be `su_image` and then this will fail. To make this as robust as possible, you'll have to get the media type source plugin information and check if its of a particular interface. I'll try to dig up an example

Reply:
Aye, and I've got a story to tell about this. It was going to be Greg but with him out today it looks like you will be getting my attention after blocker party today.